plugins/outputs/cloudrun/cloudrun.go

```diff
@@ -0,0 +1,197 @@
+package cloudrun
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/golang-jwt/jwt"
+	"github.com/influxdata/telegraf/config"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/internal"
+	"github.com/influxdata/telegraf/plugins/common/gcp"
+	"github.com/influxdata/telegraf/plugins/common/tls"
+	"github.com/influxdata/telegraf/plugins/outputs"
+	"github.com/influxdata/telegraf/plugins/serializers"
+)
+
+const (
+	defaultURL = "http://127.0.0.1:8080/telegraf"
+)
+
+var sampleConfig = `
+  ## URL is the Cloud Run Wavefront proxy address to send metrics to
+  # url = "http://127.0.0.1:8080/telegraf"
+
+  ## Timeout for Cloud Run message, suggested as 30s to account for handshaking
+  # timeout = "30s"
+
+  ## Cloud Run JSON file location
+  ## This is the location of the JSON file generated from your GCP project that's authorized to send
+  ## metrics into CloudRun.
+  ## Windows users, note that you need to use forward slashes.
+  # json_file_location = "C:/GCP/example-cr.json"
+
+  ## Data format to output.
+  ## Each data format has it's own unique set of configuration options, read
+  ## more about them here:
+  ## https://github.com/influxdata/telegraf/blob/master/docs/DATA_FORMATS_OUTPUT.md
+  # data_format = "wavefront"
+
+  ## NOTE: The default headers have already been set that is appropriate to send
+  ## metrics which are set to the following so you don't have to.
+  ## defaultContentType   = "application/octet-stream"
+  ## defaultAccept        = "application/json"
+  ## defaultMethod        = http.MethodPost
+  ## The token is generated using the URL, json_file_location, and cloudrun_email you set in your conf file
+`
+
+const (
+	defaultClientTimeout = 5 * time.Second
+	defaultContentType   = "application/octet-stream"
+	defaultAccept        = "application/json"
+	defaultMethod        = http.MethodPost
+)
+
+type CloudRun struct {
+	URL          string            `toml:"url"`
+	Timeout      config.Duration   `toml:"timeout"`
+	Headers      map[string]string `toml:"headers"`
+	JSONSecret   string            `toml:"json_file_location"`
+	ConvertPaths bool              `toml:"convert_paths"`
+	Method       string
```

Comment:
Is this a user-option? If so, please add a TOML tag and document it in `sampleConfig`. If not, do you need it exported?